File: frontend/src/types/notification.ts
// src/types/notification.ts

export interface AnalysisPayloadResponse {
  id: number;
  analysis_data: Record<string, any>;
  user_feedback_score?: number | null;
}

export interface EmailAnalysisResponse {
  id: number;
  priority_score: number;
  summary?: string;
  category?: string;
  category_id?: number;
}

export interface IMSessionStateResponse {
  priority_score: number;
  current_topic?: string;      // 💥 对齐后端真实字段
  summary_snapshot?: string;   // 💥 对齐后端真实字段
  is_read?: boolean;
}

export interface NotificationResponse {
  id: number;
  account_id: number;
  platform: 'email' | 'instagram';
  account_msg_id: string;
  status: string;

  sender?: string | null;
  external_sender_id?: string | null;
  subject?: string | null;
  cleaned_content?: string | null;

  is_from_me: boolean;
  reply_to_mid?: string | null;
  is_read?: boolean;

  email_analysis?: EmailAnalysisResponse | null;
  im_session_state?: IMSessionStateResponse | null; // 💥 核心修复：更正为真实的后端返回键名
  analysis_payload?: AnalysisPayloadResponse | null;

  received_at?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface GetNotificationsParams {
  status?: string;
  account_id?: string;
  is_read?: boolean;
  limit?: number;
  offset?: number;
}

export interface NotificationUpdatePayload {
  status?: string;
  is_read?: boolean;
}

export interface FeedbackUpdatePayload {
  user_feedback_score: number;
}

export interface GroupedNotification {
  id: string;
  platform: 'email' | 'instagram';
  account_id: number;
  external_sender_id?: string | null;
  messages: NotificationResponse[];
  latest_received_at: string;
  priority_score: number;
  title: string;
  is_read: boolean;
}

File: app/schemas/notification.py
# app/schemas/notification.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List, Dict, Any
from datetime import datetime

from app.schemas.email_analysis import EmailAnalysisResponse
from app.schemas.im_session import IMSessionStateResponse


class AnalysisPayloadResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    analysis_data: Dict[str, Any]
    user_feedback_score: Optional[int] = None


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    account_id: int
    platform: str
    account_msg_id: str
    status: str
    is_read: bool = False
    sender: Optional[str] = None
    external_sender_id: Optional[str] = None
    subject: Optional[str] = None
    cleaned_content: Optional[str] = None
    is_from_me: bool = False
    reply_to_mid: Optional[str] = None

    email_analysis: Optional[EmailAnalysisResponse] = None
    im_session_state: Optional[IMSessionStateResponse] = None

    # 💥 完美映射：直接使用底层暴露的 analysis_payload，100% 对齐你的前端 TS 接口！
    analysis_payload: Optional[AnalysisPayloadResponse] = None

    received_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class NotificationUpdate(BaseModel):
    status: Optional[str] = None
    is_read: Optional[bool] = None


class FeedbackUpdate(BaseModel):
    user_feedback_score: int = Field(..., ge=1, le=10)


# ================= Webhook 相关 =================
class InstagramReplyTo(BaseModel):
    mid: str


class InstagramMessage(BaseModel):
    mid: str
    text: Optional[str] = ""
    is_echo: bool = False
    reply_to: Optional[InstagramReplyTo] = None


class InstagramSender(BaseModel):
    id: str


class InstagramRecipient(BaseModel):
    id: str


class InstagramMessagingEvent(BaseModel):
    sender: InstagramSender
    recipient: InstagramRecipient
    timestamp: int
    message: Optional[InstagramMessage] = None


class InstagramEntry(BaseModel):
    id: str
    time: int
    messaging: List[InstagramMessagingEvent]


class InstagramWebhookPayload(BaseModel):
    object: str
    entry: List[InstagramEntry]